apps/artemis-web/pages/api/auth/login/ldap.ts

```diff
@@ -7,14 +7,22 @@ import {
 import auth from '../../../../middleware/auth';
 import { extractLdapUser } from '../../../../utils/parsers';
 import { csrf } from '../../../../libs/csrf';
+import captcha from '../../../../middleware/captcha';
 import limiter from '../../../../middleware/limiter';
+import memory from '../../../../utils/captchaMemoryStore';
 
 const handler = nc()
   .use(limiter('ldap'))
+  .use(captcha('login'))
   .use(auth)
+  .post(async (req: NextApiRequestExtended, res: NextApiResponseExtended, next) => {
+    memory.incr(req.ip);
+    next();
+  })
   .post(
     passport.authenticate('ldapauth'),
     (req: NextApiRequestExtended, res: NextApiResponseExtended, next) => {
+      memory.decr(req.ip);
```

Comment:
post-login, shouldn't the captcha failed failed try counter be set to 0? Since from now on the user succeeded?

Reply:
fair point! Let me fix that!